src/Core/Domain/Discount/Command/AddDiscountCommand.php

```diff
@@ -47,11 +55,19 @@ abstract class AddDiscountCommand
     private bool $highlightInCart = false;
     private bool $allowPartialUse = true;
     private DiscountType $type;
+    private ?DecimalNumber $percentDiscount = null;
+    private ?Money $amountDiscount = null;
+    private ?ProductId $productId = null;
+    private ?CombinationIdInterface $combinationId = null;
+    private int $reductionProduct = 0;
+    private bool $isFreeShipping = false;
```

Comment:
There's no need for this property at the moment Only a discount with type free shipping can give this action on the discount So the boolean is redundant with the type itself You can remove the property its getters/setters and focus only on `type === FREE_SHIPPING`